### ISSUE TITLE
ap-east-1 is detected, but giving AuthFailures with federated credentials

### ISSUE THREAD

**Issue by tomalok** [mcrute/alpine-ec2-ami#53]
Using my AWS account and IAM user, with ap-east-1 disabled...
```
build/resolve-profile.py test
>>> Determining region availability...-++++++++++++++++
*** WARNING: skipping disabled region ap-east-1
>>> Resolving configuration for 'v3_10-x86_64'
...
```
```
build/prune-amis.py version test
> PRUNING test/edge-x86_64 for version
* scanning: ap-east-1...
An error occurred (AuthFailure) when calling the DescribeImages operation: AWS was not able to validate the provided access credentials
* scanning: ap-northeast-1
...
```

I then enabled it on my account, wait a while, and then try again...
```
build/resolve-profile.py test
>>> Determining region availability...+++++++++++++++++
>>> Resolving configuration for 'v3_10-x86_64'
...
```
```
build/prune-amis.py version test
> PRUNING test/edge-x86_64 for version
* scanning: ap-east-1...
* scanning: ap-northeast-1...
...
```

When using this project's federated credentials, however, the results are always equivalent to when I have ap-east-1 disabled.

https://console.aws.amazon.com/billing/home/?#/account has a highlighted note about STS endpoints -- if I'm reading that right, we'll either need to activate the global STS entpoint to issue session tokens that are valid in all AWS regions, or we'll need to switch endpoints when we work with the newer regions.

**Comment by mcrute** [comment]
It looks like the issue here is how opt-in regions support STS. They have their own regional endpoints and won't accept credentials from other regions or the global region. The fix basically required a complete re-write to the identity broker but should hopefully put us in a place where better automation is supported in the future. I'm also going to have to write a release tool that will assume the proper role in the proper region, copy the AMI, and release it to the public. I've got the identity broker rewrite done and working and I've started on the release tool as well but have not finished that yet. I'm planning to wrap up this work by the end of next week and push it.

There's a very work-in-progress preview in my WIP branch if you want to get a general idea of what's happening. The only closed-source bit is the identity broker itself, which is not terribly interesting. It just uses long-lived credentials to give short-lived credentials to users.

The fix was a lot trickier than I expected but I'm most of the way done. I'll post it as a PR next week when I get it finished.
